### PR TITLE
feat: add java spec

### PR DESCRIPTION
**Pull request by Koooooo-7**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature about `Java` basic cmd.
**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for this PR, I left two comments to solve before merge.

File: src/java.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const completionSpec: Fig.Spec = {
+  name: "java",
+  description: "Launch a Java application",
+  args: {
+    name: "mainclass",
+    description: "To launch a class file",
+    template: "filepaths",
```
[the remark this answers is not in the document]
Comment (Koooooo-7) [reply src/java.ts]: I see,  yes, it should keep `cp` as a single command.

File: src/java.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      description: "Print product version to the output stream and continue",
+    },
+    {
+      name: "--dry-run",
+      description:
+        "Create VM and load main class but do not execute main method",
+    },
+    // classpath
+    {
+      name: ["--classpath", "-cp"],
```
[the remark this answers is not in the document]
Comment (Koooooo-7) [reply src/java.ts:47]: I see, yes, it should keep cp as a single command.

File: src/java.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      description: "Class search path of directories and zip/jar files",
+      args: {
+        name: "seach files",
```
Comment (fedeci) [inline src/java.ts:50]:
```suggestion
        name: "search files",
```

File: src/java.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    generators: filepaths({ extensions: ["java", "class"] }),
+  },
+  options: [
+    {
+      name: "-jar",
+      priority: 99,
+      description: "To launch the main class in a JAR file",
+      args: {
+        name: "JAR file",
+        template: "filepaths",
```
Comment (fedeci) [inline src/java.ts:20]:
```suggestion
        generators: filepaths({ extensions: ["jar"] }),
```

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**